### ISSUE TITLE
Reason for `Could not move output artifacts from sandboxed execution` is not given

### ISSUE THREAD

**Issue by Silic0nS0ldier** [bazelbuild/bazel#23283]
### Description of the bug:

I've run into an undiagnosed build issue that can be resolved by running `bazel clean` (this has manifested in our CI environments multiple times and at least once on another engineers machine).

Looking through historical issues (e.g. #13229) I can see that such internal issues previously included a stack trace showing the full exception chain. A chain which by all accounts appears to still exist behind the scenes in Bazel, its just that only the error message is shown now (meaning the cause is never shown).

Is there a way more information can be exposed to permit debugging of this issue?

### Which category does this issue belong to?

Core, Local Execution

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

As implied by this issue being undiagnosed, I've so far been unable to reproduce the issue intentionally. I have determined that CI passes if `bazel clean` is run first, which suggests a collision in the output directory.

However for investigating the lack of a stack trace, the repro for similar issues that have occurred recently should suffice. e.g. #22260 

### Which operating system are you running Bazel on?

Linux, macOS

### What is the output of `bazel info release`?

release 7.2.1

### Any other information, logs, or outputs that you want to share?

Output snippet for reference (shows 2 targets `:lib` and `:pkg` in the same package having actions fail due to outputs being unable to be copied).

```
INFO: Analyzed target //tools/protogen_regenerate:cli (414 packages loaded, 7482 targets configured).
ERROR: /___/tools/logging/BUILD.bazel:15:15: Transpiling & type-checking TypeScript project @@//tools/logging:lib [tsc -p tools/logging/tsconfig.json] failed: I/O exception during sandboxed execution: Could not move output artifacts from sandboxed execution
Target //tools/protogen_regenerate:cli failed to build
ERROR: /___/tools/logging/BUILD.bazel:8:12 Copying files to directory tools/logging/pkg failed: I/O exception during sandboxed execution: Could not move output artifacts from sandboxed execution
```

**Comment by tjgq** [comment]
Was this with or without `--verbose_failures`? We're supposed to show the stack trace when it's set, but I've seen situations where it doesn't work.

**Comment by Silic0nS0ldier** [comment]
With `--verbose_failures`.

**Comment by tjgq** [comment]
FWIW, I was wrong; there's no general expectation that `--verbose_failures` prints a stack trace (in fact, the documentation for the flag doesn't even mention it). In some contexts we do, in others we don't; some consistency would be great, but I don't want to tackle that right now. Let me see if I can improve the error message for this particular situation.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!
